### PR TITLE
ASSET MASTER R

### PR DESCRIPTION
**Pull request by mohitDehuliya**

### CODE REVIEW

File: repository/asset_repository.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/jmoiron/sqlx"
+	"github.com/vkhichar/asset-management/domain"
+)
+
+const (
+	GetAssetDetails = "SELECT id,category,status,purchase_at,purchase_cost,name,specifications FROM assets"
+)
+
+type AssetRepository interface {
+	// CreateAsset(ctx context.Context, assetCategory string, assetStatus string, purchaseCost float64, assetName string) (domain.Asset, error)
```
Comment (vkhichar) [inline repository/asset_repository.go]: Remove un necessary/ commented code from everywhere

File: handler/asset_handler.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		asset, err := asset.ListAssets(r.Context())
+
+		if err != nil {
+
+			fmt.Printf("handler:Error while Searching for Assets, %s", err.Error())
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		if err == customerrors.NoAssetsExist {
```
Comment (vkhichar) [inline handler/asset_handler.go:18]: will this code be reachable in any case?

File: service/asset_service.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// 	}
+
+// 	return &asv, nil
+// }
+
+func (service *assetService) ListAssets(ctx context.Context) ([]domain.Asset, error) {
+
+	asset, err := service.assetRepo.ListAssets(ctx)
+
+	// if err == repository.NoUserExists {
```
Comment (vkhichar) [inline service/asset_service.go]: remove commented code

File: repository/asset_repository.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/vkhichar/asset-management/domain"
+)
+
+const (
+	GetAssetDetails = "SELECT id,category,status,purchase_at,purchase_cost,name,specifications FROM assets"
```
Comment (vkhichar) [inline repository/asset_repository.go]: this constant is not used outside this package, so this should be un exported

Comment (mohitDehuliya) [reply repository/asset_repository.go]: okay I got it!

File: service/asset_service.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import (
+	"context"
+	"errors"
+
+	"github.com/vkhichar/asset-management/customerrors"
+	"github.com/vkhichar/asset-management/domain"
+	"github.com/vkhichar/asset-management/repository"
+)
+
+var ErrNoUserExists = errors.New("No asset exists here")
```
Comment (vkhichar) [inline service/asset_service.go]: @thefreeman17 remove this un used variable

File: domain/assets.go [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+type Asset struct {
+	Id             uuid.UUID       `db:"id"` //uuid package read
+	Status         string          `db:"status"`
+	Category       string          `db:"category"`
+	PurchaseAt     time.Time       `db:"purchase_at"`
+	PurchaseCost   float64         `db:"purchase_cost"`
+	Name           string          `db:"name"`
+	Specifications json.RawMessage `db:"specifications"` //encoding json read
```
Comment (wsushmita) [inline domain/assets.go]: remove comment

File: handler/asset_handler.go [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+func ListAssetHandler(asset service.AssetService) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		asset, err := asset.ListAssets(r.Context())
+		if err == customerrors.NoAssetsExist {
+			fmt.Println("handler: No assets exist")
+			w.WriteHeader(http.StatusNotFound)
+			responseBytes, _ := json.Marshal(contract.ErrorResponse{Error: "no asset found"})
```
Comment (wsushmita) [inline handler/asset_handler.go]: don't ignore error using _

File: handler/asset_handler.go [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+	"net/http"
+
+	"github.com/vkhichar/asset-management/contract"
+	"github.com/vkhichar/asset-management/customerrors"
+	"github.com/vkhichar/asset-management/service"
+)
+
+func ListAssetHandler(asset service.AssetService) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
```
Comment (wsushmita) [inline handler/asset_handler.go:15]: move set header where you write the response

File: handler/asset_handler.go [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+			responseBytes, _ := json.Marshal(contract.ErrorResponse{Error: "no asset found"})
+			w.Write(responseBytes)
+			return
+		}
+		if err != nil {
+			fmt.Printf("handler:Error while Searching for Assets, %s", err.Error())
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		w.WriteHeader(http.StatusOK)
```
Comment (wsushmita) [inline handler/asset_handler.go]: move this line after for loop

File: handler/asset_handler.go [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+		assetResp := make([]contract.Asset, 0)
+		for _, u := range asset {
+			assetResp = append(assetResp, contract.DomainToContractassets(&u))
+		}
+		responseBytes, err := json.Marshal(assetResp)
```
Comment (wsushmita) [inline handler/asset_handler.go:42]: check err != nil

**Review by wsushmita — CHANGES_REQUESTED** [review]

File: handler/asset_handler.go (hunk 8 above)
Comment (mohitDehuliya) [reply handler/asset_handler.go:15]: please elaborate

**Pull request merged**